app/models/column_preference.rb

```diff
@@ -52,4 +53,13 @@ def self.filter(default_preferences, user, action_name)
 
     default_preferences.delete(:schedules)
   end
+
+  def self.get_default_preferences(action_name, user)
+    case action_name
+    when 'products_v3_index'
+      products_v3_index_columns(user)
+    else
+      __send__("#{action_name}_columns")
+    end
+  end
```

Comment:
there is a `filter` method that seems to be doing something similar, we should put this logic in there instead of having two different method.

Reply:
Yup, I considered that method before, I think the `filter` method is serving a different purpose here. We need the method to get the `default_preferences`, which is this method.
The `filter` method takes the `default_preferences` and conditionally removes columns in it. It's _filtering_ the columns while we need all the columns, just with different visibility checks and unchecks.